src/safegdscript/script_language_safegdscript.cpp

```diff
@@ -0,0 +1,282 @@
+		"signal",
+		"super",
+		"self",
+		"true",
+		"false",
+		"null",
+		"void",
+		"bool",
+		"int",
+		"float",
+		"String",
+		"Vector2",
+		"Vector2i",
+		"Vector3",
+		"Vector3i",
+		"Vector4",
+		"Vector4i",
+		"Color",
+		"Array",
+		"Dictionary",
+		"PackedByteArray",
+		"PackedInt32Array",
+		"PackedInt64Array",
+		"PackedFloat32Array",
+		"PackedFloat64Array",
+		"PackedStringArray",
+		"PackedVector2Array",
+		"PackedVector3Array",
+		"PackedColorArray",
+		"Node",
+		"RefCounted",
+		"Resource",
+		"void",
+		"assert",
+		"await",
+		"yield",
+	};
+	return reserved_words;
+}
+bool SafeGDScriptLanguage::_is_control_flow_keyword(const String &p_keyword) const {
+	static const std::unordered_set<std::string> control_flow_keywords{
+		"if", "elif", "else", "for", "while", "match", "break", "continue", "return", "pass", "assert"
+	};
+	return control_flow_keywords.find(p_keyword.utf8().get_data()) != control_flow_keywords.end();
+}
+PackedStringArray SafeGDScriptLanguage::_get_comment_delimiters() const {
+	PackedStringArray comment_delimiters;
+	comment_delimiters.push_back("#");
+	return comment_delimiters;
+}
+PackedStringArray SafeGDScriptLanguage::_get_doc_comment_delimiters() const {
+	PackedStringArray doc_comment_delimiters;
+	doc_comment_delimiters.push_back("##");
+	return doc_comment_delimiters;
+}
+PackedStringArray SafeGDScriptLanguage::_get_string_delimiters() const {
+	PackedStringArray string_delimiters;
+	string_delimiters.push_back("\" \"");
+	string_delimiters.push_back("' '");
+	string_delimiters.push_back("\"\"\" \"\"\"");
+	return string_delimiters;
+}
+Ref<Script> SafeGDScriptLanguage::_make_template(const String &p_template, const String &p_class_name, const String &p_base_class_name) const {
+	SafeGDScript *script = memnew(SafeGDScript);
+	return Ref<Script>(script);
+}
+TypedArray<Dictionary> SafeGDScriptLanguage::_get_built_in_templates(const StringName &p_object) const {
+	return TypedArray<Dictionary>();
+}
+bool SafeGDScriptLanguage::_is_using_templates() {
+	return false;
+}
+Dictionary SafeGDScriptLanguage::_validate(const String &p_script, const String &p_path, bool p_validate_functions, bool p_validate_errors, bool p_validate_warnings, bool p_validate_safe_lines) const {
+	return Dictionary();
+}
+String SafeGDScriptLanguage::_validate_path(const String &p_path) const {
+	return String();
+}
+Object *SafeGDScriptLanguage::_create_script() const {
+	SafeGDScript *script = memnew(SafeGDScript);
+	return script;
+}
+bool SafeGDScriptLanguage::_has_named_classes() const {
+	return false;
+}
+bool SafeGDScriptLanguage::_supports_builtin_mode() const {
+	return true;
+}
+bool SafeGDScriptLanguage::_supports_documentation() const {
+	return false;
+}
+bool SafeGDScriptLanguage::_can_inherit_from_file() const {
+	return false;
+}
+bool SafeGDScriptLanguage::_can_make_function() const {
+	return false;
+}
+int32_t SafeGDScriptLanguage::_find_function(const String &p_function, const String &p_code) const {
```

Comment:
Not sure where the code is, but we can integrate the class db search function we have with the auto complete of SafeScript.